src/main/java/org/embulk/util/config/TaskMapper.java

```diff
@@ -88,6 +96,13 @@
         } catch (final IOException ex) {
             throw new UncheckedIOException("Unexpected I/O error in ObjectNode rebuilt from org.embulk.config.TaskSource.", ex);
         }
+
+        if (this.validator != null) {
+            final Set<ConstraintViolation<T>> violations = this.validator.validate(value);
+            if (!violations.isEmpty()) {
+                throw new TaskValidationException(violations);
+            }
+        }
```

Comment:
It was intentional that `TaskMapper` does not validate because `embulk-core`'s original one is implemented intentionally not to validate for `TaskSource#loadTask`.

Adding a new feature to validate is an option. But at least, the default behavior should follow the original one because many plugins would start migrating from a simple default one.

Reply:
Now I agree. I misread the old implementation and thought `TaskSource` also get validated.